src/pages/workspace/tags/WorkspaceEditTagPage.tsx

```diff
@@ -0,0 +1,114 @@
+import * as Policy from '@userActions/Policy';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+import INPUT_IDS from '@src/types/form/WorkspaceTagForm';
+import type {PolicyTagList} from '@src/types/onyx';
+
+type WorkspaceEditTagPageOnyxProps = {
+    /** All policy tags */
+    policyTags: OnyxEntry<PolicyTagList>;
+};
+
+type EditTagPageProps = WorkspaceEditTagPageOnyxProps & StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.TAG_EDIT>;
+
+function EditTagPage({route, policyTags}: EditTagPageProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const {inputCallbackRef} = useAutoFocusInput();
+
+    const validate = useCallback(
+        (values: FormOnyxValues<typeof ONYXKEYS.FORMS.WORKSPACE_TAG_CREATE_FORM>) => {
+            const errors: FormInputErrors<typeof ONYXKEYS.FORMS.WORKSPACE_TAG_CREATE_FORM> = {};
+            const tagName = values.tagName.trim();
+            const {tags} = PolicyUtils.getTagList(policyTags, 0);
+
+            if (!ValidationUtils.isRequiredFulfilled(tagName)) {
+                errors.tagName = 'workspace.tags.tagRequiredError';
+            } else if (tags?.[tagName]) {
+                errors.tagName = 'workspace.tags.existingTagError';
+            } else if ([...tagName].length > CONST.TAG_NAME_LIMIT) {
+                // Uses the spread syntax to count the number of Unicode code points instead of the number of UTF-16 code units.
+                ErrorUtils.addErrorMessage(errors, 'tagName', ['common.error.characterLimitExceedCounter', {length: [...tagName].length, limit: CONST.TAG_NAME_LIMIT}]);
+            }
+
+            return errors;
+        },
+        [policyTags],
+    );
+
+    const editTag = useCallback(
+        (values: FormOnyxValues<typeof ONYXKEYS.FORMS.WORKSPACE_TAG_CREATE_FORM>) => {
+            Policy.renamePolicyTag(route.params.policyID, {oldName: route.params.tagName, newName: values.tagName.trim()});
+            Keyboard.dismiss();
+            Navigation.goBack(ROUTES.WORKSPACE_TAGS.getRoute(route.params.policyID));
+        },
+        [route.params.policyID, route.params.tagName],
+    );
+
+    return (
+        <AdminPolicyAccessOrNotFoundWrapper policyID={route.params.policyID}>
+            <PaidPolicyAccessOrNotFoundWrapper policyID={route.params.policyID}>
+                <ScreenWrapper
+                    includeSafeAreaPaddingBottom={false}
+                    style={[styles.defaultModalContainer]}
+                    testID={EditTagPage.displayName}
+                    shouldEnableMaxHeight
+                >
+                    <HeaderWithBackButton
+                        title={translate('workspace.tags.addTag')}
+                        onBackButtonPress={Navigation.goBack}
+                    />
+                    <FormProvider
```

Comment:
Seems like we're still working on this (and I think you're on the right track given that I see WorkspaceTagForm in a few places), but just wanted to make sure that we're extracting into a component since we reuse it in create tag page, similar to what we're doing with [categories](https://github.com/Expensify/App/pull/37969/files#diff-794520d9a870a127b16ddf978c0a9a26fd91a75e51a01c2b8b6b41fc4f02b19bR1).